x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/async/AsyncTaskIndexService.java

```diff
@@ -217,9 +224,10 @@ public void updateResponse(String docId,
                                R response,
                                ActionListener<UpdateResponse> listener) {
         try {
-            final ReleasableBytesStreamOutput buffer = new ReleasableBytesStreamOutput(0, bigArrays.withCircuitBreaking());
+            final ReleasableBytesStreamOutput buffer = new ReleasableBytesStreamOutputWithLimit(
+                0, bigArrays.withCircuitBreaking(), maxResponseSize);
```

Comment:
If the size of the final search response exceeds the limit, then the initial response will stay untouched in the async-search index, and users won't see the error unless they look at the ES logs. Should we create a failure response and replace the initial response with it here?

Reply:
Addressed in 219ddd5a5a698c497eb567253d59a8f7a666f8a3